ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/ErrorFactories.scala

```diff
@@ -42,6 +42,7 @@ trait ErrorFactories {
   def unimplemented(description: String): StatusRuntimeException =
     grpcError(Status.UNIMPLEMENTED.withDescription(description))
 
+  // permission denied is intentionally without description to ensure we don't leak security relevant information by accident
```

Comment:
This is exactly the reason why the description was left out initially.

Some context: what the ledger API authorization does is only compare the content of the request (e.g., the command object) with the content of the access token, both of which are provided by the user. Ledger API authorization does not depend on any on-ledger data.

The main concern should be about leaking cryptographic secrets related to the access token (such as the correct signature or the keys used for checking the signature). If the validation/decoding of the access token fails, we only store a generic [Claims.Unauthenticated](https://github.com/digital-asset/daml/blob/main/ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala#L52) object without any details. The code responsible for sending out the gRPC error message does not have access to the original access token.

Adding details for why permission was denied should therefore be possible.

Reply:
Yeah, so @soren-da suggested to rather just leave it as is. We don't have any test ensuring that we do not leak security-relevant information through that rejection. So if we want to respond with meaningful errors, then we also need to add the whole test harness to ensure this remains secure even after refactorings. As this is out-of-scope for now, I guess that it will have to be picked up at a later point in time.